src/daemon.rs

```diff
@@ -133,6 +125,22 @@ impl Daemon {
         if blockchain_info.pruned {
             bail!("electrs requires non-pruned bitcoind node");
         }
+
+        loop {
+            if exit_flag.is_set() {
+                bail!("bitcoin RPC polling interrupted");
```

Comment:
Might be nicer if this was some specific error that top-level code can check to return success code if electrs is terminated by signal. Or at least not say in output that it's an error.